apps/test/unit/gamelab/GameLabSpriteTest.js

```diff
@@ -205,4 +205,170 @@ describe('GameLabSprite', function () {
       });
     });
   });
+
+  describe('collision types using AABBOps', function () {
+    var sprite, spriteTarget;
+
+    beforeEach(function () {
+      // sprite in to the left, moving right
+      // spriteTarget in to the right, stationary
+      sprite = createSprite(281, 100, 20, 20);
+      spriteTarget = createSprite(300, 100, 20, 20);
+      sprite.velocity.x = 3;
+      spriteTarget.velocity.x = 0;
+    });
+
+    it('stops movement of colliding sprite when sprites bounce', function () {
+      // sprite stops moving, spriteTarget moves right
+      expect(sprite.velocity.x).to.equal(3);
+      expect(spriteTarget.velocity.x).to.equal(0);
+
+      var bounce = sprite.bounce(spriteTarget);
+
+      expect(bounce).to.equal(true);
+
+      expect(sprite.position.x).to.equal(280); // move back to not overlap with spriteTarget
+      expect(spriteTarget.position.x).to.equal(300);
+      expect(sprite.velocity.x).to.equal(0);
+      expect(spriteTarget.velocity.x).to.equal(3);
```

Comment:
It would be interesting to try and get [chai-subset](http://chaijs.com/plugins/chai-subset/) working so you could say:

```javascript
expect(sprite).to.containSubset({position: {x: 280}, velocity: {x: 0}});
```

Not sure that's really more expressive though.